File: frontend/src/app/domain/work-hub/work-hub-farm-row.ts
export interface WorkHubFarmRow {
  farmId: number;
  farmName: string;
  fieldCount: number;
  totalArea: number;
  hasValidFields: boolean;
  planId: number | null;
  overdueCount: number;
  todayCount: number;
  unrecordedCount: number;
  gddDelayCount: number;
  daysExceedanceCount: number;
  thresholdExceededCount: number;
}

File: crates/agrr-domain/src/work_record/dtos/work_hub_farm_row.rs
//! Work hub farm row for `/api/v1/work/hub`.

#[derive(Debug, Clone, PartialEq)]
pub struct WorkHubFarmRow {
    pub farm_id: i64,
    pub farm_name: String,
    pub field_count: i32,
    pub total_area: f64,
    pub has_valid_fields: bool,
    pub plan_id: Option<i64>,
}
